src/main/java/rx/internal/operators/OperatorConcat.java

```diff
@@ -112,9 +112,19 @@ public void onStart() {
         }
 
         private void requestFromChild(long n) {
-            if (n <=0) return;
+            if (n <= 0) return;
             // we track 'requested' so we know whether we should subscribe the next or not
-            long previous = BackpressureUtils.getAndAddRequest(requested, n);
+            
+            final AtomicLong requestedField = requested;
+            
+            long previous;
+            
+            if (requestedField.get() != Long.MAX_VALUE) {
+                previous = BackpressureUtils.getAndAddRequest(requestedField, n);
```

Comment:
This path involves two `AtomicLong.get()`. Can't the whole block just be replaced by 
`long previous = BackpressureUtils.getAndAddRequest(requestedField, n);`?